File: src/p0012/golang/solution.go
var sd = [...]string{"", "M", "MM", "MMM"}
var sc = [...]string{"", "C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM"}
var sx = [...]string{"", "X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC"}
var si = [...]string{"", "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX"}

func intToRoman(i int) string {
	return sd[i/1000] + sc[i%1000/100] + sx[i%100/10] + si[i%10]
}

File: src/p0012/rust/solution.rs
impl Solution {
    const SD: &'static [&'static str] = &["", "M", "MM", "MMM"];
    const SC: &'static [&'static str] = &["", "C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM"];
    const SX: &'static [&'static str] = &["", "X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC"];
    const SI: &'static [&'static str] = &["", "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX"];
    pub fn int_to_roman(i: i32) -> String {
        let mut result = String::new();
        let i = i as usize;
        result.push_str(Self::SD[i / 1000]);
        result.push_str(Self::SC[i % 1000 / 100]);
        result.push_str(Self::SX[i % 100 / 10]);
        result.push_str(Self::SI[i % 10]);
        return result
    }
}
